lib/bitcoin/BitcoinClient.ts

```diff
@@ -99,18 +107,12 @@ export default class BitcoinClient {
    * @param bitcoinLockTransaction The transaction object.
    */
   public async broadcastLockTransaction (bitcoinLockTransaction: BitcoinLockTransactionModel): Promise<string> {
-    if (!(bitcoinLockTransaction.transactionObject instanceof Transaction)) {
-      throw new Error('Transaction property is not of bitcore-lib.Transaction');
-    }
-
-    const transaction: Transaction = bitcoinLockTransaction.transactionObject;
-    const rawTransaction = transaction.serialize();
 
-    return this.broadcastTransactionRpc(rawTransaction);
+    return this.broadcastTransactionRpc(bitcoinLockTransaction.serializedTransactionObject);

```

Comment:
Why do we need this? To hide the BitcoinLockTransactionModel's implementation detail?

Reply:
We want to rebroadcast the full transaction for the following scenario:
1. The service created a lock txn.
1. It saved the info to the DB
1. The service crashed before it could broadcast.
1. The service comes back up, sees the txn in the DB but not on the blockchain. So it just rebroadcast.

We don't want to broadcast-first-and-then-save. Because if the service crashes before saving then we don't know that there's a lock and may lose that info.